app/src/main/java/com/wizeline/heroes/ui/adapters/CharacterRecyclerViewAdapter.kt

```diff
@@ -19,17 +20,31 @@ class CharacterRecyclerViewAdapter(
     private val list: ArrayList<Character>,
     private val context: Context
 ) :
-    RecyclerView.Adapter<CharacterRecyclerViewAdapter.ViewHolder>() {
+    ListAdapter<Character, CharacterRecyclerViewAdapter.ViewHolder>(DiffCallback()) {
+
+    private class DiffCallback : DiffUtil.ItemCallback<Character>() {
+        override fun areItemsTheSame(oldItem: Character, newItem: Character): Boolean {
+            return oldItem.name == newItem.name
```

Comment:
I would compare the items with the 'id' instead of name. Because that seems to be the unique identifier.

Reply:
Done.